contracts/pcv/angle/GaugeLens.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.10;  
+
+import "../../metagov/utils/LiquidityGaugeManager.sol";
+import "../IPCVDepositBalances.sol";
+
+/// @title GaugeLens
+/// @author Fei Protocol
+/// @notice a contract to read tokens held in a gauge
+contract GaugeLens is IPCVDepositBalances {
+
+    /// @notice the gauge inspected
+    address public immutable gaugeAddress;
+
+    /// @notice the address of the contract staking in the gauge
+    address public immutable stakerAddress;
+
+    /// @notice the token the lens reports balances in
+    address public immutable override balanceReportedIn;
+
+    constructor(
+        address _gaugeAddress, 
+        address _stakerAddress
+    ) {
+        gaugeAddress = _gaugeAddress;
+        stakerAddress = _stakerAddress;
+        balanceReportedIn = ILiquidityGauge(_gaugeAddress).staking_token();
+    }
+
+    function balance() public view override returns(uint256) {
+        return ILiquidityGauge(gaugeAddress).balanceOf(stakerAddress);
+    }
+
+    function resistantBalanceAndFei() public view override returns(uint256, uint256) {
```

Comment:
"resistant" balance means that it can't be manipulated with a flashloan or something transitory like that

I'll add natspec

The lens reports the amount of tokens staked in a gauge, and reports in this token. Unless the token staked in the gauge is FEI, the FEI balance will always be zero. And there are no gauges where naked FEI is staked, it's only LP tokens & stuff like that. The lens simply reports the number of LP tokens in that case. And there is a second "uniswap lens" that, given a balance of LP tokens, deduces a number of tokens & fei held within them.